src/index.rs

```diff
@@ -0,0 +1,247 @@
+//! Represents the type of an [`Index`] in ArangoDB. The following types are
+//! supported:
+//! * Fulltext
+//! * Geo
+//! * Hash
+//! * Persistent
+//! * Skiplist
+//! * Ttl (Time to live)
+//!
+//! An index of type [`Primary`] cannot be created and is only available for
+//! the retrieval of existing indexes, as ArangoDB creates a primary index on every
+//! collection.
+//! For detailed information about ArangoDB indexes, please check out the official
+//! ArangoDB [documentation](https://www.arangodb.com/docs/stable/http/indexes.html).
+use serde::{Deserialize, Serialize};
+
+/// Represents the type of an [`Index`]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum Type {
+    Primary,
+    Fulltext,
+    Geo,
+    Hash,
+    Persistent,
+    Skiplist,
+    Ttl,
+}
+
+/// Represents an index on a collection in ArangoDB.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Index {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    is_newly_created: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    selectivity_estimate: Option<u8>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    error: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    code: Option<u16>,
+    fields: Vec<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    name: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    unique: Option<bool>,
+    r#type: Type,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    sparse: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    min_length: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    deduplicate: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    geo_json: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    expire_after: Option<u32>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    in_background: Option<bool>,
+}
+
+/// Represents a collection of indexes on a collection in ArangoDB.
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct IndexCollection {
+    pub error: bool,
+    pub code: u16,
+    pub indexes: Vec<Index>,
+}
+
+/// Response from ArangoDB when deleting an index
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DeleteIndexResponse {
+    pub id: String,
+    pub error: bool,
+    pub code: u16,
+}
+
+impl Index {
+    fn default(r#type: Type) -> Self {
```

Comment:
I prefer implementing `Default` trait in this case.

Reply:
I think it's because he can't randomy pick the index type, the default is relative to the index type here (`r#type` argument)